### DISCUSSION TITLE
Deployment Error - Check for Broken Link

### DISCUSSION THREAD

**Discussion by adityamajumdar** [alshedivat/al-folio#2721] [Q&A] [answered]
# Method 1: Commit and Push to Github
After I commit and push the changes to my github, I keep getting an error which says check for broken links on site. I have no links on my website as of now and when I check the error message this is what it says (please refer to the attached image.) 
<img width="1134" alt="Screenshot 2024-09-25 at 2 09 02 PM" src="https://github.com/user-attachments/assets/795f08d3-b372-4620-b0a3-41c3da1ed5f3">

After I open one of the messages this is what it says: 
<img width="1141" alt="Screenshot 2024-09-25 at 2 09 58 PM" src="https://github.com/user-attachments/assets/0874c53d-6f02-4e13-99b8-a0eb4e3849ed">

# Method 2: Manual Deployment 
When I tried to manually deploy the website using the actions tab, this is the error I got

<img width="1130" alt="Screenshot 2024-09-25 at 2 11 26 PM" src="https://github.com/user-attachments/assets/d059102a-f8a2-432a-b640-e0d75e4593aa">

I've followed the instructions in the INSTALL.md and also looked at the FAQs but I'm still not able to resolve this error. Can someone please assist me with this??

**Comment by george-gca** [comment]
If you check our [FAQ](https://github.com/alshedivat/al-folio/blob/master/FAQ.md#what-do-all-these-github-actionsworkflows-mean), it states:

- `broken-links-site.yml`: checks for broken links in your built website with the [lychee-action](https://github.com/lycheeverse/lychee-action)
- `broken-links.yml`: checks for broken links in your repository with the [lychee-action](https://github.com/lycheeverse/lychee-action)

Note that none of these block your site from being published, this is useful only to inform you if any links are broken.

Now, your errors are unexpectedly not useful. What changes have you made to your site already?

**Reply by adityamajumdar** [reply]
I've only made simple changes, like I changed the about.md, added the news posts and my resume's pdf that's all. It runs on localhost properly but after I commit and push it doesnt deploy the website. Not sure where I went wrong? 
if needed this the repository for my website: https://github.com/adityamajumdar/adityamajumdar.github.io

**Reply by george-gca** [reply]
2 things:
- you must do small incremental changes to your repository, so it is easier to pinpoint when it breaks. [This commit](https://github.com/adityamajumdar/adityamajumdar.github.io/commit/9f1d8477f0922694d466811098236ddce9e00bb1#diff-ecec67b0e1d7e17a83587c6d27b6baaaa133f42482b07bd3685c77f34b62d883), for example, changes too many files in one single commit, so it would be harder to identify which change caused the error
- checking the [deploy action in your repo](https://github.com/adityamajumdar/adityamajumdar.github.io/actions/workflows/deploy.yml) you can see in which commit the build started to fail (which is the one above). The problem here is that you commited your `Gemfile.lock`.
 
We are currently having some discussions about this error with apple silicon users in #2705. Check if doing `git restore Gemfile.lock` is enough to fix your case. If not, I'd recommend downloading again the [Gemfile.lock](https://github.com/alshedivat/al-folio/blob/master/Gemfile.lock) from our repo, replacing your local one, and pushing this to your repo. Then, try to do `git update-index --assume-unchanged Gemfile.lock` to keep your `Gemfile.lock` changes only in your local computer.

**Reply by adityamajumdar** [reply]
Oh I see! I'll delete the repo and try to do it again then and keep you posted with the results!

**Reply by adityamajumdar** [reply]
Okay so I deleted my repo and redid all the initial steps. I cloned the repo and changed just the first and last name in the config.yml file. There was still a deployment error. I've attached the image below: 
<img width="1512" alt="Screenshot 2024-09-25 at 4 17 18 PM" src="https://github.com/user-attachments/assets/7807e6fc-7c58-4665-b194-005ea327087a">

<img width="1497" alt="Screenshot 2024-09-25 at 4 17 43 PM" src="https://github.com/user-attachments/assets/3a04c508-2c7c-4f1f-ad02-6c4ccf43b2a0">

Not sure what's going wrong here? 
link to github: https://github.com/adityamajumdar/adityamajumdar.github.io

**Reply by george-gca** [reply, marked as answer]
The problem here is that you commited your `Gemfile.lock`.
 
We are currently having some discussions about this error with apple silicon users in #2705. Check if doing `git restore Gemfile.lock` is enough to fix your case. If not, I'd recommend downloading again the [Gemfile.lock](https://github.com/alshedivat/al-folio/blob/master/Gemfile.lock) from our repo, replacing your local one, and pushing this to your repo. Then, try to do `git update-index --assume-unchanged Gemfile.lock` to keep your `Gemfile.lock` changes only in your local computer.

**Reply by adityamajumdar** [reply]
I tried the first method but it didn't work. The second method did work tho! The site is deploying properly and I can see the edits, thank you!! :D

**Reply by george-gca** [reply]
Idk why when running locally on a Mac it overwrites the Gemfile.lock. We are still figuring this out.
